snapcraft/internal/remote_build/_launchpad.py

```diff
@@ -44,28 +45,72 @@
 class LaunchpadClient:
     """Launchpad remote builder operations."""
 
-    def __init__(self, *, project: Project, build_id: str, user: str) -> None:
+    def __init__(
+        self,
+        *,
+        project: Project,
+        build_id: str,
+        user: str,
+        architectures: Optional[List[str]] = None,
+        git_branch: str = "master",
+        core18_channel: str = "stable",
+        snapcraft_channel: str = "edge",
```

Comment:
TODO: reuse https://github.com/snapcore/snapcraft/blob/master/snapcraft/internal/build_providers/_snap.py#L40 for similar experience on setup

Reply:
Good to know!